### ISSUE TITLE
Cannot register user with long email

### ISSUE THREAD

**Issue by anxolerd** [uaprom-summer-2015/Meowth#179]
## Description
It is impossible to register a user who has a long (longer than 30 character length) e-mail address due to model restrictions (see [[here]](https://github.com/uaprom-summer-2015/Meowth/blob/master/project/models.py#L71))

## STR
1. Start app
2. Go to admin to create a new user
3. Enter email that is longer than 30 characters *(for example [my](https://github.com/anxolerd) email is longer than 30 characters, so it is a real case)*
4. Try to save this user

## Expected result
**Result 1** User was successfully created and received an email about his account was created; A corresponding DB entry was created.
**Result 2** There was an error during user creation, and user did not receive any email. No DB entry was created.

## Actual result
- 500 error returned with the following stacktrace:
```pytb
[2015-08-23 20:07:16 +0300] [6127] [ERROR] Error handling request
Traceback (most recent call last):
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
     context)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
     cursor.execute(statement, parameters)
 psycopg2.DataError: value too long for type character varying(30)
 
 
 The above exception was the direct cause of the following exception:
 
 Traceback (most recent call last):
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/gunicorn/workers/sync.py", line 130, in handle
     self.handle_request(listener, req, client, addr)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/gunicorn/workers/sync.py", line 171, in handle_request
     respiter = self.wsgi(environ, resp.start_response)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1836, in __call__
     return self.wsgi_app(environ, start_response)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1820, in wsgi_app
     response = self.make_response(self.handle_exception(e))
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1403, in handle_exception
     reraise(exc_type, exc_value, tb)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
     raise value
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1817, in wsgi_app
     response = self.full_dispatch_request()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1477, in full_dispatch_request
     rv = self.handle_user_exception(e)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1381, in handle_user_exception
     reraise(exc_type, exc_value, tb)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
     raise value
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1475, in full_dispatch_request
     rv = self.dispatch_request()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/app.py", line 1461, in dispatch_request
     return self.view_functions[rule.endpoint](**req.view_args)
   File "/home/anxolerd/workspace/Meowth/project/auth/decorators.py", line 20, in wrapped
     return func(*args, **kwargs)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/views.py", line 84, in view
     return self.dispatch_request(*args, **kwargs)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/flask/views.py", line 149, in dispatch_request
     return meth(*args, **kwargs)
   File "/home/anxolerd/workspace/Meowth/project/admin/utils.py", line 56, in post
     self.model.bl.create(form.data)
   File "/home/anxolerd/workspace/Meowth/project/bl/auth.py", line 30, in create
     model.bl.save()
   File "/home/anxolerd/workspace/Meowth/project/bl/utils.py", line 64, in save
     db.session.commit()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/scoping.py", line 150, in do
     return getattr(self.registry(), name)(*args, **kwargs)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 790, in commit
     self.transaction.commit()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 392, in commit
     self._prepare_impl()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 372, in _prepare_impl
     self.session.flush()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 2004, in flush
     self._flush(objects)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 2122, in _flush
     transaction.rollback(_capture_exception=True)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
     compat.reraise(exc_type, exc_value, exc_tb)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/util/compat.py", line 182, in reraise
     raise value
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/session.py", line 2086, in _flush
     flush_context.execute()
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/unitofwork.py", line 373, in execute
     rec.execute(self)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/unitofwork.py", line 532, in execute
     uow
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/persistence.py", line 174, in save_obj
     mapper, table, insert)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/orm/persistence.py", line 779, in _emit_insert_statements
     execute(statement, params)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 914, in execute
     return meth(self, multiparams, params)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
     return connection._execute_clauseelement(self, multiparams, params)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
     compiled_sql, distilled_params
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
     context)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
     exc_info
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/util/compat.py", line 188, in raise_from_cause
     reraise(type(exception), exception, tb=exc_tb, cause=exc_value)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/util/compat.py", line 181, in reraise
     raise value.with_traceback(tb)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
     context)
   File "/home/anxolerd/workspace/Meowth/env/lib/python3.4/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
     cursor.execute(statement, parameters)
sqlalchemy.exc.DataError: (psycopg2.DataError) value too long for type character varying(30)
20:07:16 web.1    |  [SQL: 'INSERT INTO users (login, password, name, surname, email, role) VALUES (%(login)s, %(password)s, %(name)s, %(surname)s, %(email)s, %(role)s) RETURNING users.id'] [parameters: {'role': 0, 'login': 'pinkiepie', 'name': 'Pinkie', 'password': 'pbkdf2:sha1:1000$QSsGGsBv$78491ee3046404cc3ec1923ba86d69e787acb19d', 'surname': 'Pie', 'email': 'my.very.long.test.email.address.goes.here@gmail.com'}]

```
![500](https://cloud.githubusercontent.com/assets/6533582/9429304/34bc8c46-49d4-11e5-8f4d-256fdfec6690.png)

- Email about user creation was sent:
![email](https://cloud.githubusercontent.com/assets/6533582/9429356/bde68b24-49d5-11e5-8c11-24a3141f2bba.png)

- No DB entry was created:
![db](https://cloud.githubusercontent.com/assets/6533582/9429361/e8735214-49d5-11e5-872b-bb1af7c1cafe.png)

## Suggestions
Edit model and create corresponding schema and (if needed) data migration.
See [[standard]](http://tools.ietf.org/html/rfc5321.html#section-4.5.3.1) -- 255 symbols for domain + 64 symbols for username + 1 `@` symbol = **320 symbols**.

**Comment by akhalyavka** [comment]
Look #183

**Comment by Cybran111** [comment]
Fixed in #184
